apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -536,32 +536,50 @@ map_call_get(Data, State) ->
     [MapId]   = get_args([word], Data),
     {KeyType, _ValType} = aevm_eeevm_maps:map_type(MapId, State),
     [_, KeyPtr]  = get_args([word, word], Data),
-    {ok, KeyBin} = aevm_eeevm_state:heap_to_binary(KeyType, KeyPtr, State),
-    Res = case aevm_eeevm_maps:get(MapId, KeyBin, State) of
-            false -> aeso_data:to_binary(none);
-            <<ValPtr:256, ValBin/binary>> ->
-                %% Some hacky juggling to build an option value.
-                NewPtr = 32 + byte_size(ValBin),
-                <<NewPtr:256, ValBin/binary, 1:256, ValPtr:256>>
-          end,
-    {ok, {ok, Res}, 0, State}.
+    case aevm_eeevm_state:heap_to_binary(KeyType, KeyPtr, State) of
+        {ok, KeyBin, GasUsed} ->
+            Res = case aevm_eeevm_maps:get(MapId, KeyBin, State) of
+                    false -> aeso_data:to_binary(none);
+                    <<ValPtr:256, ValBin/binary>> ->
+                        %% Some hacky juggling to build an option value.
+                        NewPtr = 32 + byte_size(ValBin),
+                        <<NewPtr:256, ValBin/binary, 1:256, ValPtr:256>>
+                  end,
+            {ok, {ok, Res}, GasUsed, State};
+        {error, _} = Err ->
+            {ok, Err, aevm_eeevm_state:gas(State), State}
```

Comment:
I am not convinced about accessing gas in VM state here. `Gas` first argument of `aevm_ae_primops:call/4` is not the same as VM gas - it is capped and with stipend (and may be different depending on value operand of call). Checking.

Reply:
I see [even if computation led to negative number no error is caused](https://github.com/aeternity/epoch/blob/c4742b1697ab7d1ec18a85ed4ab03a1f795605aa/apps/aevm/src/aevm_eeevm.erl#L1508) and [gas is set to zero anyway because of `{error, _}`](https://github.com/aeternity/epoch/blob/c4742b1697ab7d1ec18a85ed4ab03a1f795605aa/apps/aevm/src/aevm_eeevm.erl#L1531), so this has no functional impact - only decreased readability. Ok.